From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Thu, 11 Jun 2026 20:06:22 -0400
Subject: [PATCH 56/59] clang-mg: Update CIndex loop for arm declarations


diff --git a/clang/tools/libclang/CIndex.cpp b/clang/tools/libclang/CIndex.cpp
index d80d53b34f06..74fb2921f88c 100644
--- a/clang/tools/libclang/CIndex.cpp
+++ b/clang/tools/libclang/CIndex.cpp
@@ -1347,7 +1347,7 @@ bool CursorVisitor::VisitCXXMGConditionalMemberDecl(
         Visit(MakeCXCursor(Arm.Condition, StmtParent, TU, RegionOfInterest)))
       return true;
 
-    for (Decl *SubD : Arm.Decls) {
+    for (Decl *SubD : D->getArmDecls(Arm)) {
       if (!SubD)
         continue;
       if (Visit(MakeCXCursor(SubD, TU, RegionOfInterest)))
-- 
2.54.0.windows.1

